src/github.com/matrix-org/dendrite/clientapi/routing/directory.go

```diff
@@ -113,6 +113,23 @@ func SetLocalAlias(
 		}
 	}
 
+	// Check that the alias does not fall within an exclusive namespace of an
+	// application service
+	for _, appservice := range cfg.Derived.ApplicationServices {
+		for key, namespaceMaps := range appservice.NamespaceMap {
+			if key == "aliases" {
```

Comment:
why are you looping over a map to find a key ?
```go
namespaces, ok := appservice.NamespaceMap["aliases"]
```
Then check `if ok`

Reply:
Change, thanks. I think I'll be doing a separate PR to be removing the whole NamespaceMap->[]Namespace relationship, as app services aren't ever, for instance, going to have more than one "users" namespace.